### DISCUSSION TITLE
Keep stack between eval

### DISCUSSION THREAD

**Discussion by sasq64** [mruby/mruby#5500] [Q&A] [answered]
I am implementing a REPL and want to keep local variables between calls to eval. I've looked at mirb and tried to do what it does,  but I get a crash,  maybe because I'm initiating the call from ruby instead of C, so I guess I call vm exec recursively ?

I also looked at eval.c but it's more complex (maybe because it needs to work when called from ruby).

Any help appreciated.

**Comment by matz** [comment]
Show me the code, please.

**Comment by matz** [comment, marked as answer]
Probably you can use `mruby-binding` gem (#5362, #5491).

```ruby
b = binding
b.eval "a=12"
b.eval "p a"     # => 12
```

Don't forget to include `mruby-binding` gem when build.

**Reply by sasq64** [reply]
Is the binding gem tested? I get very weird bugs when I use it. When assigning variables they sometimes end up containing different objects from random places?

**Reply by sasq64** [reply]
Oh, it seems that it was a bug that was recently fixed; I pulled latest and now it seems to work :)

**Comment by sasq64** [comment]
`binding` of course! Thank you.
